File: integer-to-english-words.cs
public class Solution {
    
        private static Dictionary<int, string> trials;
    
            public Solution() {
        trials = new Dictionary<int, string> {
            {1000000000, "{0} Billion {1}"},
            {1000000, "{0} Million {1}"},
            {1000, "{0} Thousand {1}"},
            {100, "{0} Hundred {1}"},
            {90, "Ninety {1}"},
            {80, "Eighty {1}"},
            {70, "Seventy {1}"},
            {60, "Sixty {1}"},
            {50, "Fifty {1}"},
            {40, "Forty {1}"},
            {30, "Thirty {1}"},
            {20, "Twenty {1}"},
            {19, "Nineteen"},
            {18, "Eighteen"},
            {17, "Seventeen"},
            {16, "Sixteen"},
            {15, "Fifteen"},
            {14, "Fourteen"},
            {13, "Thirteen"},            
            {12, "Twelve"},
            {11, "Eleven"},
            {10, "Ten"},
            {9, "Nine"},
            {8, "Eight"},
            {7, "Seven"},
            {6, "Six"},
            {5, "Five"},
            {4, "Four"},
            {3, "Three"},
            {2, "Two"},
            {1, "One"},
            {0, "Zero"}
        };
    }


    public string NumberToWords(int num) {
        
foreach(var pair in trials)
{
    if(num <=10 && num==pair.Key)
    {
        return pair.Value;
    }
    
    int n = num / pair.Key;
    num = num % pair.Key;
    if(n >=1)
    {
        if(num==0)
        {
            return String.Format(pair.Value,NumberToWords(n),"").Trim();
        }
        else
        {
            return String.Format(pair.Value,NumberToWords(n),NumberToWords(num));
        }
    }
}
        return "";
        
        
        
        
        
        
    }
}

File: Integer-to-english-words.java
// Time Complexity : O(1) = for each triplet calling helper trice.

// Space Complexity : O(1)

// Did this code successfully run on Leetcode : YES

// Appoarch: Using helper function and decoding each triplet and using the 3 different arrays
// print the string.

// 273. Integer to English Words

class Solution {
    String[] below_20 = {"", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight",
    "Nine", "Ten", "Eleven", "Twelve", "Thirteen","Fourteen","Fifteen",
    "Sixteen","Seventeen","Eighteen","Nineteen"};
    String[] tens = {"","Ten","Twenty","Thirty","Forty","Fifty","Sixty","Seventy","Eighty",
    "Ninety"};
    String[] thousands = {"", "Thousand","Million","Billion"};
    public String numberToWords(int num) {
        String result = "";
        int i = 0;
        if(num == 0) return "Zero";
        while(num > 0){
            int triplet = num%1000;
            if(triplet != 0){
                result = helper(triplet).trim() + " " + thousands[i] + " " + result;
            }
            num = num/1000;
            i++;
        }
        return result.trim();
    }
    private String helper(int num){
        if(num < 20){
            return below_20[num];
        }
        else if(num < 100){
            return tens[num/10] + " " + helper(num%10);
        }
        else{
            return below_20[num/100] + " Hundred " + helper(num%100);
        }
    }
}